src/unix/cygwin.c

```diff
@@ -35,10 +35,69 @@ int uv_uptime(double* uptime) {
   return 0;
 }
 
-int uv_resident_set_memory(size_t* rss) {
-  /* FIXME: read /proc/meminfo? */
-  *rss = 0;
+int uv_resident_set_memory(size_t* rss)
+{
+  char buf[1024];
+  const char* s;
+  ssize_t n;
+  long val;
+  int fd;
+  int i;
+  int e;
+  struct sysinfo si;
+
+  do
+    fd = open("/proc/self/stat", O_RDONLY);
+  while (fd == -1 && errno == EINTR);
+
+  if (fd == -1)
+    return UV__ERR(errno);
+
+  do
+    n = read(fd, buf, sizeof(buf) - 1);
+  while (n == -1 && errno == EINTR);
+
+  uv__close(fd);
+  if (n == -1)
+    return UV__ERR(errno);
+  buf[n] = '\0';
```

Comment:
```suggestion
  rc = uv__slurp("/proc/self/stat", buf, sizeof(buf));
  if (rc < 0)
    return rc;
```
Zero-terminates the buffer.